### PR TITLE
fix: make revalidation synchronous

### LINKED ISSUE #596: revalidatenode: Connection to NAST Closed Prematurely
The connection to NAST closed before we could fully process the node that needed to be revalidated

image

### PR DESCRIPTION
**Pull request by maxsarratt**
resolves #596

### CODE REVIEW

File: pkg/event/base_publisher.go [hunk 1]
```diff
@@ -54,3 +56,26 @@ func (p *publisher) Publish(data interface{}) {
 	msg, _ := json.Marshal(data)
 	_, _ = p.client.PublishAsync(string(p.subject), msg, p.ackHandler)
 }
+
+// PublishSync sends a message to the designated subject using a synchronous approach.
+func (p *publisher) PublishSync(data interface{}) error {
+	if os.Getenv("ENV") == "test" {
```
Comment (TingSyuanWang) [inline pkg/event/base_publisher.go:62]: Hi, Why do we need to check the environment here?

**Review by TingSyuanWang — APPROVED** [review]
Once we figured out why do we need to check the os.Getenv, we're good to go.

File: pkg/event/base_publisher.go (hunk 1 above)
Comment (maxsarratt) [reply pkg/event/base_publisher.go:62]: I think it is an old setup that requires updates; otherwise, the unit tests will fail.

Comment (TingSyuanWang) [reply pkg/event/base_publisher.go:62]: Thank you for the clarification!

**Pull request merged**